GitHub workflow action failures for large feeds

I've noticed an increased incidence of GitHub workflow action failures ([link](https://github.com/MobilityData/gtfs-validator/actions?query=is%3Afailure)) for our `end_to_end` and `acceptance_test` workflows.  Digging deeper into these failures, the validator itself isn't failing.  Instead, our workflow runners are being terminated with the following message:

> The runner has received a shutdown signal. This can happen when the runner service is stopped, or a manually started runner is canceled.

The process itself includes the following in its logs:

> Error: Process completed with exit code 143.

This is consistent with the process being terminated externally via SIGTERM.

So what's actually going on?  Other folks are reporting the [same issue](https://github.com/actions/runner-images/issues/6709) to GitHub and it's pointed out that most of these runners are using `ubuntu-latest`, which has been upgraded to `22.04` over the past week ([link](https://github.com/actions/runner-images/issues/6399)).  That potentially changes the resource envelope of the running worker, which might lead to resource exhaustion at different points than the previous runner.

Indeed, looking at our action failures, it seems to occur most often for large feeds such as `no-unknown-agder-kollektivtrafikk-as-gtfs-1078` (11M stop-time entries) and `de-unknown-ulmer-eisenbahnfreunde-gtfs-1081` (huge shapes.txt file).

So what do we do about this?  Still digging into that.

But just to check, @aababilov and @asvechnikov2, your recent changes over the past month should have reduced memory usage for large feeds, not increased it, correct?

--- a/.github/workflows/acceptance_test.yml
+++ b/.github/workflows/acceptance_test.yml
@@ -127,10 +127,9 @@ jobs:
       matrix: ${{ steps.set-matrix.outputs.matrix }}
   get-reports:
     needs: [ fetch-urls, pack-master, pack-snapshot ]
-    runs-on: 
-      # We use machines with more memory to run validation, as large feeds
-      # can consume too much heap for default machine instances (see #1304).
-      labels: ubuntu-latest-4-cores
+    # We use machines with more memory to run validation, as large feeds
+    # can consume too much heap for default machine instances (see #1304).
+    runs-on: ubuntu-latest-4-cores
     strategy:
       matrix: ${{ fromJson(needs.fetch-urls.outputs.matrix) }}
     steps:

--- a/.github/workflows/acceptance_test.yml
+++ b/.github/workflows/acceptance_test.yml
@@ -130,7 +130,7 @@ jobs:
     runs-on: 
       # We use machines with more memory to run validation, as large feeds
       # can consume too much heap for default machine instances (see #1304).
-      group: ubuntu-latest-4-cores
+      label: ubuntu-latest-4-cores
     strategy:
       matrix: ${{ fromJson(needs.fetch-urls.outputs.matrix) }}
     steps:

--- a/.github/workflows/acceptance_test.yml
+++ b/.github/workflows/acceptance_test.yml
@@ -127,7 +127,10 @@ jobs:
       matrix: ${{ steps.set-matrix.outputs.matrix }}
   get-reports:
     needs: [ fetch-urls, pack-master, pack-snapshot ]
-    runs-on: ubuntu-latest
+    runs-on: 
+      # We use machines with more memory to run validation, as large feeds
+      # can consume too much heap for default machine instances (see #1304).
+      group: ubuntu-latest-4-cores
     strategy:
       matrix: ${{ fromJson(needs.fetch-urls.outputs.matrix) }}
     steps:

--- a/.github/workflows/acceptance_test.yml
+++ b/.github/workflows/acceptance_test.yml
@@ -130,7 +130,7 @@ jobs:
     runs-on: 
       # We use machines with more memory to run validation, as large feeds
       # can consume too much heap for default machine instances (see #1304).
-      label: ubuntu-latest-4-cores
+      labels: ubuntu-latest-4-cores
     strategy:
       matrix: ${{ fromJson(needs.fetch-urls.outputs.matrix) }}
     steps:

--- a/.github/workflows/end_to_end.yml
+++ b/.github/workflows/end_to_end.yml
@@ -82,10 +82,9 @@ jobs:
       matrix: ${{ steps.set-matrix.outputs.matrix }}
   run-on-data:
     needs: [ fetch-urls, pack-snapshot ]
-    runs-on: 
-      # We use machines with more memory to run validation, as large feeds
-      # can consume too much heap for default machine instances (see #1304).
-      labels: ubuntu-latest-4-cores
+    # We use machines with more memory to run validation, as large feeds
+    # can consume too much heap for default machine instances (see #1304).
+    runs-on: ubuntu-latest-4-cores
     strategy:
       matrix: ${{ fromJson(needs.fetch-urls.outputs.matrix) }}
     steps:

--- a/.github/workflows/end_to_end.yml
+++ b/.github/workflows/end_to_end.yml
@@ -85,7 +85,7 @@ jobs:
     runs-on: 
       # We use machines with more memory to run validation, as large feeds
       # can consume too much heap for default machine instances (see #1304).
-      label: ubuntu-latest-4-cores
+      labels: ubuntu-latest-4-cores
     strategy:
       matrix: ${{ fromJson(needs.fetch-urls.outputs.matrix) }}
     steps:

--- a/.github/workflows/end_to_end.yml
+++ b/.github/workflows/end_to_end.yml
@@ -85,7 +85,7 @@ jobs:
     runs-on: 
       # We use machines with more memory to run validation, as large feeds
       # can consume too much heap for default machine instances (see #1304).
-      group: ubuntu-latest-4-cores
+      label: ubuntu-latest-4-cores
     strategy:
       matrix: ${{ fromJson(needs.fetch-urls.outputs.matrix) }}
     steps:

--- a/.github/workflows/end_to_end.yml
+++ b/.github/workflows/end_to_end.yml
@@ -82,7 +82,10 @@ jobs:
       matrix: ${{ steps.set-matrix.outputs.matrix }}
   run-on-data:
     needs: [ fetch-urls, pack-snapshot ]
-    runs-on: ubuntu-latest
+    runs-on: 
+      # We use machines with more memory to run validation, as large feeds
+      # can consume too much heap for default machine instances (see #1304).
+      group: ubuntu-latest-4-cores
     strategy:
       matrix: ${{ fromJson(needs.fetch-urls.outputs.matrix) }}
     steps:

